nucypher/blockchain/eth/utils.py

```diff
@@ -62,3 +69,107 @@ def get_tx_cost_data(transaction_dict: TxParams):
     max_cost_wei = max_unit_price * transaction_dict["gas"]
     max_cost = Web3.from_wei(max_cost_wei, "ether")
     return max_cost, max_price_gwei, tx_type
+
+
+def rpc_endpoint_health_check(endpoint: str, max_drift_seconds: int = 60) -> bool:
+    """
+    Checks the health of an Ethereum RPC endpoint by comparing the timestamp of the latest block
+    with the system time. The maximum drift allowed is `max_drift_seconds`.
+    """
+    query = {
+        "jsonrpc": "2.0",
+        "method": "eth_getBlockByNumber",
+        "params": ["latest", False],
+        "id": 1,
+    }
+    LOGGER.debug(f"Checking health of RPC endpoint {endpoint}")
+    try:
+        response = requests.post(
+            endpoint,
+            json=query,
+            headers={"Content-Type": "application/json"},
+            timeout=5,
+        )
```

Comment:
Any particular reason for doing this instead of creating a one-off web3 instance?

Reply:
No, not really. It just got my attention when I saw a line of code parsing hexstrings into ints to get the block timestamp, which is something that web3py handles automatically via the default pythonic middleware.